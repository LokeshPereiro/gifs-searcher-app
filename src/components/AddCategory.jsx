import { useState } from "react";
import PropTypes from "prop-types";

export function AddCategory({ onNewSearchingGif }) {
  const [inputValue, setInputValue] = useState("");

  const onInputChange = (e) => {
    setInputValue(e.target.value);
  };

  const onSubmittedValue = (e) => {
    // console.log("Hola Mundo desde el OnSubmit Testing!");

    e.preventDefault();
    // console.log(inputValue);
    // Pequeña validacion para que se introducza más de 1 caracter en el input y que no se envie input vacio
    if (inputValue.trim().length <= 1) return;
    // onNewSearchingGif((prevSearchedKeys) => [inputValue, ...prevSearchedKeys]);

    setInputValue("");
    onNewSearchingGif(inputValue.trim());
  };

  return (
    <>
      <form onSubmit={onSubmittedValue} aria-label="form">
        <input
          type="text"
          placeholder="Search Your Fav Gifs"
          onChange={onInputChange}
          value={inputValue}
        />
      </form>
    </>
  );
}

AddCategory.propTypes = {
  onNewSearchingGif: PropTypes.func.isRequired,
};
