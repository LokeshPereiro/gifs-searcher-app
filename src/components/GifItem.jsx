import PropTypes from "prop-types";

export function GifItem({ title, url }) {
  // console.log(title, url);
  return (
    <div className="card">
      <img src={url} alt={title} />
      <h4>{title}</h4>
    </div>
  );
}

GifItem.propTypes = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
};
