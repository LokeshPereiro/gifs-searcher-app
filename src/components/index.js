export * from "./AddCategory";
export * from "./GifGridItems";
export * from "./GifItem";
