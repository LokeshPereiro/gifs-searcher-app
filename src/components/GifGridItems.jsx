import { GifItem } from "../components/GifItem";
import { useFetchGifs } from "../hooks/useFetchGifs";
import PropTypes from "prop-types";

export function GifGridItems({ searchItems }) {
  const { gifs, isLoading } = useFetchGifs(searchItems);
  // console.log(isLoading);
  // console.log(gifs);
  return (
    <>
      <h3>{searchItems}</h3>
      {isLoading && <h4 style={{ color: "red" }}>GIFs Loading...</h4>}
      <div className="card-grid">
        {gifs.map((gifObj) => (
          // <li key={id}>{title}</li>
          <GifItem key={gifObj.id} {...gifObj} />
        ))}
      </div>
    </>
  );
}

GifGridItems.propTypes = {
  searchItems: PropTypes.string.isRequired,
};
