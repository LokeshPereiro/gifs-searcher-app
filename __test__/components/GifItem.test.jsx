import { render, screen } from "@testing-library/react";
import { GifItem } from "../../src/components/GifItem";

describe("Testing <GifItem />", () => {
  const title = "Demon Slayer";
  const url = "https://demon-slayer.com/kimetsu.jpg";

  test("Hacer MATCH con el SNAPSHOT", () => {
    const { container } = render(<GifItem title={title} url={url} />);
    expect(container).toMatchSnapshot();
  });

  test("Mostrar la imagen con el URL y ALT indicado", () => {
    render(<GifItem title={title} url={url} />);
    // screen.debug();

    // expect(screen.getByRole("img").src).toBe(url);
    // destructuring del elemento img
    const { src, alt } = screen.getByRole("img");
    expect(src).toBe(url);
    expect(alt).toBe(alt);
  });

  test("Segurarnos de que se muestre el TITULO en el component", () => {
    render(<GifItem title={title} url={url} />);
    // Hay dos prop 'title', no podemos quitar ningun ya que son PROPS, luego el Snapshot tambi√©n falla ya que hubo cambio
    expect(screen.getByText(title)).toBeTruthy();
  });
});
