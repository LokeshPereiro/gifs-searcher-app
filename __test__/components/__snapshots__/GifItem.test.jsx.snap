// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <GifItem /> Hacer MATCH con el SNAPSHOT 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Demon Slayer"
      src="https://demon-slayer.com/kimetsu.jpg"
    />
    <h4>
      Demon Slayer
    </h4>
  </div>
</div>
`;
